sequencer/src/l1_client.rs

```diff
@@ -136,21 +139,20 @@ impl L1Client {
         &self,
         prev_finalized: Option<u64>,
         new_finalized: u64,
-        address: Address,
     ) -> Vec<FeeInfo> {
         // `prev` should have allready been processed unless we
         // haven't processed *any* blocks yet.
         let prev = prev_finalized.map(|prev| prev + 1).unwrap_or(0);
         // No new blocks have been finalized, therefore there are no
         // new deposits.
-        if prev == new_finalized {
+        if prev >= new_finalized {
```

Comment:
This is also problematic, I think. Take `prev_finalized = Some(1)` and `new_finalized = 2`. In this case we _should_ process new deposits (from block 2), but we won't because `prev == 2`. I think the check we want is something like `prev_finalized == Some(new_finalized)`

Reply:
Also we should probably have some tests that fail if this check is incorrect